assets/app/scripts/controllers/deployments.js

```diff
@@ -73,6 +94,20 @@ angular.module('openshiftConsole')
       });
     }
 
+    function associateRunningDeploymentToDeploymentConfig(deploymentsByDeploymentConfig) {
+      var deploymentConfigDeploymentsInProgress = {};
+      angular.forEach(deploymentsByDeploymentConfig, function(deploymentConfigDeployments, deploymentConfigName) {
+        deploymentConfigDeploymentsInProgress[deploymentConfigName] = {};
+        angular.forEach(deploymentConfigDeployments, function(deployment, deploymentName) {
+          var deploymentStatus = $scope.deploymentStatus(deployment);
+          if (deploymentStatus === "New" || deploymentStatus === "Pending" || deploymentStatus === "Running") {
+            deploymentConfigDeploymentsInProgress[deploymentConfigName][deploymentName] = deployment;
+          }
+        });
+      });
+      return deploymentConfigDeploymentsInProgress;
+    }
+
```

Comment:
Wondering if we may want functions like this moved up into a service.  It may be handy to have available in other contexts & keep the controllers light-weight.  We don't have many services in front of `DataService`, but perhaps it makes sense to have a wrapper.  `Data.by()` provides `deployments.by("metadata.name")`, if we wrapped it we could then also have methods like `deployments.associateRunningDeploymentToDeploymentConfig()` on the service.  We might be able to shorten names then, perhaps something like `deployments.byConfig(config)`.

In the call to `watches.push(DataService.watch("replicationcontrollers", $scope, function(deployments) {});`, the `deployments` object could look something like:

```javascript
// deployments
return {
  data: { /* the request data */ },
  // functions we can share so they don't have to live in controllers
  api: {
    byConfig: function() { /* */ },
    foo: function() { /* */ }
  }
}
```